Models/EventTypesListSortFilterProxyModel.cpp

```diff
@@ -0,0 +1,24 @@
+#include "EventTypesListSortFilterProxyModel.h"
+#include "EventTypesListModel.h"
+
+
+EventTypesListSortFilterProxyModel::EventTypesListSortFilterProxyModel(QObject* parent)
+    : QSortFilterProxyModel(parent) {}
+
+bool EventTypesListSortFilterProxyModel::filterAcceptsRow(int source_row, const QModelIndex& /*source_parent*/) const {
+  return sourceModel()->data(sourceModel()->index(source_row, 0), Qt::UserRole).toBool();
+}
+
+bool EventTypesListSortFilterProxyModel::lessThan(const QModelIndex& source_left,
+                                                  const QModelIndex& source_right) const {
+  //Try to sort alpha but keep grouped events together
```

Comment:
That's known as a stable sort.
https://en.wikipedia.org/wiki/Category:Stable_sorts